Fix miscellaneous bugs, add custom timeout

kincaidoneil opened the pull request.

sharafian commented:
What did you use to recompile this? And have you tested to make sure it works?

michielbdejong commented:
https://ethereum.github.io/browser-solidity/ is what I used, it's an online Solidity compiler. You can click 'Contract details (bytecode, interface etc.)' on there to get the Web3 deploy commands.

I checked that this works for https://rinkeby.etherscan.io/address/0x8B3fBD781096b51e68448C6e5B53B240f663199F, which is the contract instance I'm using on the ethereum testnet, but probably, fixing this for my contract instance will break it for your contract instance?

sharafian commented:
yeah, if the ABI changes it won't be able to use the instance on the livenet anymore. If that version is working, I don't see a compelling reason to change it

michielbdejong commented:
ok, I'll try to deploy a contract with the same ABI as the one on the live net.

michielbdejong commented:
This is [the ABI deployed and used on the testnet](https://rinkeby.etherscan.io/address/0x8B3fBD781096b51e68448C6e5B53B240f663199F#code). 
 But I had some trouble using etherscan's Verify And Publish function, so I think I'll end up deploying a new contract at a new address at some point anyway, and then I can also use the one from master.

So yeah, I'll keep the branch, but the PR can be closed!

michielbdejong commented:
btw, ping me on Slack if you want me to add your ssh key to the testnet bootstrap server, it has a synced full geth node connected to Rinkeby, might be useful in case you want to deploy a different contract to there, so people can connect to the Interledger-testnet-of-testnets with the latest ethereum plugin :)

kincaidoneil committed:
Print correct source of debug output

Instead of "...asym-client," it should be "...asym-server"

kincaidoneil committed:
feat: enable custom timeout

kincaidoneil committed:
fix: custom data handler throws error

kincaidoneil committed:
fix: don't send 0 amount paychan updates

- It's unnecessary
- Receiver may not understand why it's not greater than existing claim
- There was a related bug in the server plugin

kincaidoneil committed:
Rename clientChannel var for clarification

- By contrast, currentChannel is an object with channel details
- The two are otherwise easier to confuse

kincaidoneil committed:
fix: can't send money, receiver is undefined

Causes frustrating cascading Web3 errors, e.g. Solidity invalid number of arguments

kincaidoneil committed:
Use single quotes everywhere

kincaidoneil committed:
Add default timeout of 30s

emschwartz commented on index.js:
```diff
@@ -231,7 +239,7 @@ class Plugin extends PluginMiniAccounts {
     const primary = data.protocolData[0]
     if (primary.protocolName === 'machinomy') {
       const payment = JSON.parse(primary.data.toString())
-      console.log("GOT PAYMENT", payment)
```
We should remove console.log statements. Could make it a `debug` one if you think it's useful to have this type of information printed

emschwartz commented:
Since there aren't that many changes here it might be worth squashing them into a single commit

kincaidoneil committed:
Remove unnecessary console.log
